python/ml4ir/base/config/parse_args.py

```diff
@@ -21,30 +25,31 @@ def define_args(self):
         self.add_argument(
             "--data_format",
             type=str,
+            choices=DataFormatKey.get_all_keys(),
             default="tfrecord",
             help="Format of the data to be used. "
-            "Should be one of the Data format keys in ml4ir/config/keys.py",
+            "Supported Data Formats  are specified in ml4ir/base/config/keys.py",
         )
 
         self.add_argument(
             "--tfrecord_type",
             type=str,
+            choices=TFRecordTypeKey.get_all_keys(),
             default="example",
             help="TFRecord type of the data to be used. "
-            "Should be one of the TFRecord type keys in ml4ir/config/keys.py",
+            "Supported TFRecord type are specified in ml4ir/base/config/keys.py",
         )
 
         self.add_argument(
             "--feature_config",
             type=str,
-            default=None,
             help="Path to YAML file or YAML string with feature metadata for training.",
         )
 
         self.add_argument(
             "--model_file",
             type=str,
-            default="",
+            default=None,
```

Comment:
should we have "required" args, that don't allow empty string or None? 

cc: @lastmansleeping

Reply:
setting this to `not-required` @jakemannix as in the pipeline docstring: 
```
model_file : str, optional
            Path to pretrained model file to be loaded for evaluation or retraining
```

Reply:
@jakemannix if you have some time, please take a look at this PR.